extensions/amp-ad-network-doubleclick-impl/0.1/amp-ad-network-doubleclick-impl.js

```diff
@@ -561,15 +552,48 @@ export class AmpAdNetworkDoubleclickImpl extends AmpA4A {
     // TODO: Check for required and allowed parameters. Probably use
     // validateData, from 3p/3p/js, after noving it someplace common.
     const startTime = Date.now();
-    const rtcRequestPromise = isExperimentOn(this.win, 'disable-rtc') ?
-    Promise.resolve({}) : this.executeRtc_();
-    return rtcRequestPromise.then(rtcResult => {
+    return opt_rtcResponsesPromise.then(rtcResponseArray => {
+      const rtcParams = this.mergeRtcResponses_(rtcResponseArray);
       return googleAdUrl(
           this, DOUBLECLICK_BASE_URL, startTime, Object.assign(
-              this.getBlockParameters_(), rtcResult, PAGE_LEVEL_PARAMS_));
+              this.getBlockParameters_(), rtcParams, PAGE_LEVEL_PARAMS_));
     });
   }
 
+  /**
+   * Merges all of the rtcResponses into the JSON targeting and
+   * category exclusions.
+   * @param {?Array<rtcResponseDef>} rtcResponseArray
+   * @private
+   */
+  mergeRtcResponses_(rtcResponseArray) {
+    if (!rtcResponseArray) {
+      return null;
+    }
+    let rtcParams = {};
+    const artc = [];
+    const ati = [];
+    const ard = [];
+    rtcResponseArray.forEach(rtcResponse => {
+      artc.push(rtcResponse.rtcTime);
+      ati.push(!rtcResponse.error ? RTC_ATI_ENUM.RTC_SUCCESS :
+               RTC_ATI_ENUM.RTC_FAILURE);
+      ard.push(rtcResponse.callout);
+      rtcParams = Object.assign(rtcParams, rtcResponse.response);
+    });
+    ['targeting', 'categoryExclusions'].forEach(key => {
+      if (rtcParams[key]) {
+        this.jsonTargeting_[key] =
+            !!this.jsonTargeting_[key] ?
+            deepMerge(this.jsonTargeting_[key], rtcParams[key]) :
+            rtcParams[key];
+      }
+    });
+    return {'artc': artc.join() || null, 'ati': ati.join(), 'ard': ard.join(),
```

Comment:
Given we're reporting EVERY error, I'm concerned this could be an unboundedly long parameter value.  Either we should only report for results where there was a call-out (e.g. do not include non-secure URLs, duplicates, missing vendor, etc) or you should limit the length of the array.  I suggest we only send errors for actual requests which will limit the size to 5.

Also what do you plan to do for SRA?  The targeting merge will work fine because there is code that combines blocks for targeting but you'll lose the error/timing values unless you add explicit support.